src/components/combo/combo.ts

```diff
@@ -0,0 +1,654 @@
+
+  /**
+   * The placeholder attribute of the search input.
+   * @attr placeholder-search
+   */
+  @property({ attribute: 'placeholder-search', type: String })
+  public placeholderSearch = 'Search';
+
+  /** The direction attribute of the control. */
+  @property({ reflect: true })
+  public override dir: 'ltr' | 'rtl' | 'auto' = 'auto';
+
+  /** Sets the open state of the component. */
+  @property({ type: Boolean })
+  public open = false;
+
+  /** @hidden @internal */
+  @property({ type: Boolean })
+  public flip = true;
+
+  /** @hidden @internal */
+  @property({ type: Boolean, attribute: 'same-width' })
+  public sameWidth = true;
+
+  /**
+   * The key in the data source used when selecting items.
+   * @attr value-key
+   */
+  @property({ attribute: 'value-key', reflect: false })
+  public valueKey?: Keys<T>;
+
+  /**
+   * The key in the data source used to display items in the list.
+   * @attr display-key
+   */
+  @property({ attribute: 'display-key', reflect: false })
+  public displayKey?: Keys<T> = this.valueKey;
+
+  /**
+   * The key in the data source used to group items in the list.
+   * @attr group-key
+   */
+  @property({ attribute: 'group-key', reflect: false })
+  public groupKey?: Keys<T> = this.displayKey;
+
+  /**
+   * Sorts the items in each group by ascending or descending order.
+   * @attr group-sorting
+   * @type {"asc" | "desc"}
+   */
+  @property({ attribute: 'group-sorting', reflect: false })
+  public groupSorting: GroupingDirection = 'asc';
+
+  /**
+   * An object that configures the filtering of the combo.
+   * @attr filtering-options
+   * @type {FilteringOptions<T>}
+   * @param filterKey - The key in the data source used when filtering the list of options.
+   * @param caseSensitive - Determines whether the filtering operation should be case sensitive.
+   */
+  @property({
+    attribute: 'filtering-options',
+    reflect: false,
+    converter: filteringOptionsConverter,
+  })
+  public filteringOptions: FilteringOptions<T> = {
+    filterKey: this.displayKey,
+    caseSensitive: false,
+  };
+
+  /**
+   * Enables the case sensitive search icon in the filtering input.
+   * @attr case-sensitive-icon
+   */
+  @property({ type: Boolean, attribute: 'case-sensitive-icon', reflect: false })
+  public caseSensitiveIcon = false;
+
+  /**
+   * Disables the filtering of the list of options.
+   * @attr disable-filtering
+   */
+  @property({ type: Boolean, attribute: 'disable-filtering', reflect: false })
+  public disableFiltering = false;
+
+  /**
+   * The template used for the content of each combo item.
+   * @type {(item: T) => TemplateResult}
+   */
+  @property({ attribute: false })
+  public itemTemplate: (item: ComboRecord<T>) => TemplateResult = (item) => {
+    if (typeof item !== 'object' || item === null) {
+      return String(item) as any;
+    }
+
+    if (this.displayKey) {
+      return html`${item[this.displayKey]}`;
+    }
+
+    return html`${String(item)}`;
```

Comment:
```suggestion
    return html`${this.displayKey ? item[this.displayKey] : item}`
```
Isn't that the same, or is there some reason for this code?